packages/formation-react/src/components/modal/Modal.jsx

```diff
@@ -29,9 +30,9 @@ class Modal extends React.Component {
   }
 
   setupModal() {
-    this.applyFocusToModal();
+    this.applyFocusToFirstModalElement();
     document.body.classList.add('modal-open');
-    document.addEventListener('keyup', this.handleDocumentKeyUp, false);
+    document.addEventListener('keydown', this.handleDocumentKeyDown, false);
```

Comment:
switching to a `keydown` here. `keydown` is triggered _before_ `focus` while `keyup` is triggered _after_ `focus` and we want to take some action before the browser has focused on the next element.